# Bisulfite sequencing (BS-seq)

BS-seq a sequencing techniques that aims to detect methylation in genomic DNA, specifically methylated cytosines (5mC). Treated with sodium bisulfite and then seuqnced, methylation of cytosine at signle-base resoultion level can be captured against untreated results.

Upon bisulphite treatment, unmethylated cytosines (C) are converted to uracils (U) while the methylated cytosine (5mC) remain unchanged (see figure below). In the untreated samples, the cytosines (C) are read as thumidines (T) while the 5mC read as cytosines (C). The differences between treated and untreated groups determine the location of methylated cytosines.

## Overview

<p align="center">
<img width="300" height="auto" src="https://github.com/sta426hs2022/week02-technology-overview-row2left/blob/main/Bisulfite-sequencing-DNA-is-denatured-and-treated-with-sodium-bisulfite-and-then-PCR-is.png">
</p>

## Applications for Bisulfite sequencing

* Compare differentially methylated loci from different samples (ex.: healthy controls versus patients with cancer)

* Discover correlations between cytosine methylation and transcriptional regulation

* Discover inherited methylations in nearly any organism

* Analyse collective effects of Methylation on organisms


## Limitations Bisulfite sequencing

* Bisulfite sequencing can't discriminate between 5-methylcytosine and 5-hydroxymethylcytosine which is another possible DNA methylation in mammalian DNA. This means that the output from bisulfite sequencing can no longer be defined as solely DNA methylation.

*  If conversion is incomplete of the unmethylated cytosine residue to uracil, it will incorrectly interpret the unconverted unmethylated cytosines as methylated cytosines, resulting in false positive results.

* Degradation of DNA during bisulfite treatment


## Links and references

Li, Y., & Tollefsbol, T. O. (2011). DNA methylation detection: bisulfite genomic sequencing analysis. _Methods in molecular biology (Clifton, N.J.)_, 791, 11–21. https://doi.org/10.1007/978-1-61779-316-5_2

Berman, B. P., Weisenberger, D. J., Aman, J. F., Hinoue, T., Ramjan, Z., Liu, Y., Noushmehr, H., Lange, C. P., van Dijk, C. M., Tollenaar, R. A., Van Den Berg, D., & Laird, P. W. (2011). Regions of focal DNA hypermethylation and long-range hypomethylation in colorectal cancer coincide with nuclear lamina-associated domains. _Nature genetics_, 44(1), 40–46. https://doi.org/10.1038/ng.969

Feil, R., Charlton, J., Bird, A. P., Walter, J., & Reik, W. (1994). Methylation analysis on individual chromosomes: improved protocol for bisulphite genomic sequencing. _Nucleic acids research_, 22(4), 695–696. https://doi.org/10.1093/nar/22.4.695

Prcture Reference: Study of Tissue-Specific CpG Methylation of DNA in Extended Genomic Loci - Scientific Figure on ResearchGate. Available from: https://www.researchgate.net/figure/Bisulfite-sequencing-DNA-is-denatured-and-treated-with-sodium-bisulfite-and-then-PCR-is_fig2_7793389 [accessed 26 Sep, 2022]

## Group Members

[daggermaster3000](https://github.com/daggermaster3000), [wyq977](https://github.com/wyq977), [sarloet](https://github.com/sarloet)

